front-page.php

```diff
@@ -114,6 +114,7 @@
           </label>
         </div>
       </div>
+      <input type="hidden" name="referer" value="<?php echo $_SERVER['HTTP_REFERER']; ?>" />
```

Comment:
Yeahh that's what I'm thinking too. I might expand on this and throw this in a cookie